subprojects/core-api/src/main/java/org/gradle/api/internal/Generated.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal;
+
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Annotation to remove Gradle generated code from JaCoCo.
```

Comment:
:x: JaCoCo is an example use case but other tools could leverage this for other means. I would make that documentation more generic by explaining what code should be annotated with it instead of what tools would consume it.

Reply:
a suggestion would be most appreciated